storage/register.go

```diff
@@ -0,0 +1,30 @@
+package storage
+
+import (
+	"io"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// Register defines methods for the register index.
+type Register interface {
+	RegisterReader
+	RegisterWriter
+	io.Closer
+}
+
+// RegisterReader defines read-only operations on the register index.
+type RegisterReader interface {
+	// Get register by the register ID at a given block height.
+	//
+	// If the register at the given height was not indexed, returns the highest
+	// height the register was indexed at.
+	// An error is returned if the register was not indexed at all.
```

Comment:
Also define any expected sentinels (errors that the caller should consider benign and handle gracefully)
```suggestion
	// Get register by the register ID at a given block height.
	//
	// If the register at the given height was not indexed, returns the highest
	// height the register was indexed at.
	// An error is returned if the register was not indexed at all. Expected errors:
	// - storage.ErrNotFound if the register was not found in the db
```

see the error api section in the [CodingConventions](https://github.com/onflow/flow-go/blob/master/CodingConventions.md#best-practice-guidelines) for reference